airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/DefaultJobPersistence.java

```diff
@@ -570,14 +579,68 @@ public Optional<Job> getLastSyncJob(final UUID connectionId) throws IOException
         .fetch(BASE_JOB_SELECT_AND_JOIN + WHERE +
             "CAST(jobs.config_type AS VARCHAR) = ? " + AND +
             "scope = ? " +
-            "ORDER BY jobs.created_at DESC LIMIT 1",
+            ORDER_BY_JOB_CREATED_AT_DESC + LIMIT_1,
             Sqls.toSqlName(ConfigType.SYNC),
             connectionId.toString())
         .stream()
         .findFirst()
         .flatMap(r -> getJobOptional(ctx, r.get(JOB_ID, Long.class))));
   }
 
+  /**
+   * For each connection ID in the input, find that connection's latest sync job and return it in an
+   * Optional if one exists.
+   */
+  @Override
+  public List<Optional<Job>> getLastSyncJobsForConnections(final List<UUID> connectionIds) throws IOException {
+    if (connectionIds.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return jobDatabase.query(ctx -> ctx
+        .fetch("SELECT DISTINCT ON (scope) * FROM jobs "
+            + WHERE + "CAST(jobs.config_type AS VARCHAR) = ? "
+            + AND + scopeInList(connectionIds)
+            + "ORDER BY scope, created_at DESC",
+            Sqls.toSqlName(ConfigType.SYNC),
+            connectionIds.stream().map(UUID::toString).map(Names::singleQuote).collect(Collectors.joining(",")))
+        .stream()
+        .map(r -> getJobOptional(ctx, r.get("id", Long.class)))
+        .collect(Collectors.toList()));
+  }
+
+  /**
+   * For each connection ID in the input, find that connection's most recent non-terminal sync job and
+   * return it in an Optional if one exists.
+   */
+  @Override
+  public List<Optional<Job>> getRunningSyncJobForConnections(final List<UUID> connectionIds) throws IOException {
+    if (connectionIds.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return jobDatabase.query(ctx -> ctx
+        .fetch("SELECT DISTINCT ON (scope) * FROM jobs "
+            + WHERE + "CAST(jobs.config_type AS VARCHAR) = ? "
+            + AND + scopeInList(connectionIds)
+            + AND + JOB_STATUS_IS_NON_TERMINAL
+            + "ORDER BY scope, created_at DESC",
+            Sqls.toSqlName(ConfigType.SYNC),
+            connectionIds.stream().map(UUID::toString).map(Names::singleQuote).collect(Collectors.joining(",")),
+            JobStatus.NON_TERMINAL_STATUSES.stream().map(Sqls::toSqlName).map(Names::singleQuote).collect(Collectors.joining(",")))
```

Comment:
same thought here on assigning these to variables.

again might just be me - I try not to use the fluent api style too liberally as it can be tough to read when unbroken.

Reply:
Similar situation here, I actually meant to delete this, I already moved this into a constant called `JOB_STATUS_IS_NON_TERMINAL` but this line was left-over